collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -312,29 +323,22 @@ public void recordBean(
         // attrDescription tends not to be useful, so give the fully qualified name too.
         String help = attrDescription + " (" + beanName + attrName + ")";
 
-        String attrNameSnakeCase = snakeCasePattern.matcher(attrName).replaceAll("$1_$2").toLowerCase();
+        String attrNameSnakeCase = toSnakeCase(attrName);
+
+        // Evict patterns no longer listed in the rules
+        // TODO
```

Comment:
When yaml gets updated, patterns could be removed; I'll work on this if there in interested in this overall change.